Orange/widgets/data/owtransform.py

```diff
@@ -74,21 +81,36 @@ def set_data(self, data):
     @Inputs.template_data
     @check_sql_input
     def set_template_data(self, data):
-        self.template_domain = data and data.domain
+        self.template_data = data
 
     def handleNewSignals(self):
+        summary, details, kwargs = self.info.NoInput, "", {}
+        if self.data or self.template_data:
+            n_data = len(self.data) if self.data else 0
+            n_template = len(self.template_data) if self.template_data else 0
+            summary = f"{self.info.format_number(n_data)}, " \
+                      f"{self.info.format_number(n_template)}"
+            kwargs = {"format": Qt.RichText}
+            details = format_multiple_summaries([
+                ("Data", self.data),
+                ("Template data", self.template_data)
+            ])
+        self.info.set_input_summary(summary, details, **kwargs)
         self.apply()
 
     def apply(self):
         self.clear_messages()
         transformed_data = None
-        if self.data and self.template_domain is not None:
+        if self.data and self.template_data is not None:
```

Comment:
If self.template_data is not None, it evaluates to True (the is not None part is not necessary).